pkg/lib/cockpit-po-plugin.js

```diff
@@ -1,115 +1,136 @@
+
+const config = {};
+
+function get_po_files() {
+    try {
+        const linguas_file = path.resolve(config.srcdir, "po/LINGUAS");
+        const linguas = fs.readFileSync(linguas_file, 'utf8').match(/\S+/g);
+        return linguas.map(lang => path.resolve(config.srcdir, 'po', lang + '.po'));
+    } catch (error) {
+        if (error.code !== 'ENOENT') {
+            throw error;
+        }
 
-const srcdir = process.env.SRCDIR || '.';
+        /* No LINGUAS file?  Fall back to globbing.
+         * Note: we won't detect .po files being added in this case.
+         */
+        return glob.sync(path.resolve(config.srcdir, 'po/*.po'));
+    }
+}
 
-export default class {
-    constructor(options) {
-        if (!options)
-            options = {};
-        this.subdir = options.subdir || '';
-        this.wrapper = options.wrapper || 'cockpit.locale(PO_DATA);';
+function get_plural_expr(statement) {
+    try {
+        /* Check that the plural forms isn't being sneaky since we build a function here */
+        Jed.PF.parse(statement);
+    } catch (ex) {
+        console.error("bad plural forms: " + ex.message);
+        process.exit(1);
     }
 
-    get_po_files(compilation) {
-        try {
-            const linguas_file = path.resolve(srcdir, "po/LINGUAS");
-            const linguas = fs.readFileSync(linguas_file, 'utf8').match(/\S+/g);
-            compilation.fileDependencies.add(linguas_file); // Only after reading the file
-            return linguas.map(lang => path.resolve(srcdir, 'po', lang + '.po'));
-        } catch (error) {
-            if (error.code !== 'ENOENT') {
-                throw error;
-            }
+    const expr = statement.replace(/nplurals=[1-9]; plural=([^;]*);?$/, '(n) => $1');
+    if (expr === statement) {
+        console.error("bad plural forms: " + statement);
+        process.exit(1);
+    }
+
+    return expr;
+}
 
-            /* No LINGUAS file?  Fall back to globbing.
-             * Note: we won't detect .po files being added in this case.
-             */
-            return glob.sync(path.resolve(srcdir, 'po/*.po'));
+function buildFile(po_file, webpack_module, webpack_compilation) {
+    if (webpack_compilation)
+        webpack_compilation.fileDependencies.add(po_file);
+
+    return new Promise((resolve, reject) => {
+        const parsed = gettext_parser.po.parse(fs.readFileSync(po_file), 'utf8');
+        delete parsed.translations[""][""]; // second header copy
+
+        const rtl_langs = ["ar", "fa", "he", "ur"];
+        const dir = rtl_langs.includes(parsed.headers.language) ? "rtl" : "ltr";
+
+        // cockpit.js only looks at "plural-forms" and "language"
+        const chunks = [
+            '{\n',
+            ' "": {\n',
+            `  "plural-forms": ${get_plural_expr(parsed.headers['plural-forms'])},\n`,
+            `  "language": "${parsed.headers.language}",\n`,
+            `  "language-direction": "${dir}"\n`,
+            ' }'
```

Comment:
Offtopic: Should this at one point become a properly serialized json object with JSON.stringify.

Reply:
Yes, maybe -- I hope that nothing (or not much) depends on the precise formatting of this blob. Happy to do that as a follow-up!